strawberryfields/configuration.py

```diff
@@ -14,139 +14,231 @@
 r"""
 This module contains the :class:`Configuration` class, which is used to
 load, store, save, and modify configuration options for Strawberry Fields.
+
+Configuration options
+*********************
+
+.. note::
+    The following configuration options are taken into consideration:
+
+        * **authentication_token (str)** (*required*): the token used for user authentication
+        * **hostname (str)** (*optional*): the name of the host to connect to
+        * **use_ssl (bool)** (*optional*): specifies if requests should be sent using SSL
+        * **port (int)** (*optional*): the port to be used when connecting to the remote service
+        * **debug (bool)** (*optional*): determines if the debugging mode is requested
+
+Environment variables
+*********************
+
+.. note::
+
+    When loading the configuration, the following environment variables are checked:
+
+    * SF_API_AUTHENTICATION_TOKEN
+    * SF_API_HOSTNAME
+    * SF_API_USE_SSL
+    * SF_API_DEBUG
+    * SF_API_PORT
 """
-import os
 import logging as log
+import os
 
 import toml
 from appdirs import user_config_dir
 
 log.getLogger()
 
+BOOLEAN_KEYS = {"debug": False, "use_ssl": True}
+INTEGER_KEYS = {"port": 443}
+
+class ConfigurationError(Exception):
+    """Exception used for configuration errors"""
+
+
+def load_config(filename="config.toml", **kwargs):
+    """Load configuration from keyword arguments, configuration file or
+    environment variables.
+
+    .. note::
+
+        The configuration dictionary will be created based on the following
+        (order defines the importance, going from most important to least
+        important):
+
+        1. keyword arguments passed to ``load_config``
+        2. data contained in environmental variables (if any)
+        3. data contained in a configuration file (if exists)
+
+    Kwargs:
+        filename (str): the name of the configuration file to look for
+
+        Furthermore configuration options as detailed in
+        :mod:`strawberryfields.configuration`
```

Comment:
This is great 🙂 From a maintainability point of view, there is a single place which is the source of truth for allowed configuration options!